### ISSUE TITLE
"Dimension time already exists" error when opening aerosol dataset

### ISSUE THREAD

**Issue by HelenClifton** [CCI-Tools/cate#634]
### Expected behavior

@forman, @JanisGailis Executing use case 22 happy path with cate GUI
Downloaded dataset esacci.AEROSOL.mon.L3C.AER_PRODUCTS.AATSR.Envisat.ORAC.03-02.r1
Expected it to open

### Actual behavior

An error message is seen

![uc22_time_error](https://user-images.githubusercontent.com/29569434/39532824-1d666576-4e1d-11e8-8e7f-f619dc2ac554.PNG)
Traceback here: 
[uc22_time_error.txt](https://github.com/CCI-Tools/cate/files/1967890/uc22_time_error.txt)

Notes:
- From the cate CLI, the dataset will be downloaded without error (command **cate ds copy**) but the error occurs when it is opened (command **cate res open**)
- I was using a clean install of cate-2.0.0-dev.10 (not the auto-updated version)
- It may be related to #623

### Steps to reproduce the problem

Select data store: ESA CCI Open Data Portal
Highlight data source: esacci.AEROSOL.mon.L3C.AER_PRODUCTS.AATSR.Envisat.ORAC.03-02.r1
Click on “Download and/or open remote dataset”
In “Download Data Source” window:
•	Select Time Constraint and enter start time of 2010-01-01 and end time of 2010-12-31
•	Select Region Constraint, enter:
o	Lon. from:  -100
o	Lon. from: 80
o	Lat. from: -90
o	Lat. to: 90
•	Select Variables Constraint, select variable AOD550_mean from the list 
•	Select “Download and make local data source“ and enter Unique identifier UC22_aerosol
•	Click on “Download & Open Local”

### Specifications 

Windows 7 Professional
cate-2.0.0-dev.10

**Comment by forman** [comment]
@papesci, it is [this piece of code](https://github.com/CCI-Tools/cate/blob/39bb4e7799be190a1900378d1d0ed4fb173110de/cate/core/ds.py#L628-L629) that I marked [with a TODO in my branch](https://github.com/CCI-Tools/cate/blob/20b88c82b92cfef3da7339b2c2a1b23d0ba87324/cate/core/ds.py#L592-L595) that causes this. Obviously @kbernat added these lines once to fix the "missing time dimension" problem but instead of really fixing it, we now have a "dead" time dimension in all of our localized datasets without the actual `time` coordinate variable in there. Aaargh!  👎 

@papesci, @JanisGailis we have to review how we normalize datasets opened by `open_dataset` in the future. I am strongly voting for doing any normalisation by default, including adding a missing time coordinate variable / dimension plus, plus correcting (see #620) and renaming lat/lon coordinate variables / dimensions. The current situation is less than unsatisfying, we have operations `open_dataset`, `normalize`, `adjust_spatial_attrs`, `adjust_temporal_attrs`, `fix_lon_360` that all do some kind of normalisation (see also #555 for single grid cells with lat/lon dimensions retained).

**Comment by papesci** [comment]
Yes indeed, we have to support the user as much as possible making easier its work. This mean also executing automatically all the phases preparatory to access data. 
By the way we have to deal also with dataset that are not normalizable, for example DS with no lat or log dimension, in this case i think the system should   always open the dataset with a warning notifying  the user his about to access a DS with a unrecognized structure.

**Comment by forman** [comment]
@JanisGailis I agree as long as you are referring to software design - compose generic `normalize()` operation from of smaller specialized operations each useful on its own. That's absolutely fine. Then `normalize()` should have a set of kwargs that control the behaviour, e.g. `gen_missing_time_dim=True`.

However, I was referring to the individual normalisation steps that should happen in Cate's `open_dataset()` so that a maximum number of operations can be applied without forcing users to add intermediate normalisation steps. By default, `open_dataset()` should apply the most common normalisations that we are aware off so that we have a guarantee that if there are time coordinates, there is a 1-D variable `"time"` of type `datetime64` with the dimension `"time"`. And if there are longitude coordinates, there is a 1-D variable `"lon"` etc and that corresponding CF attributes are set correctly.

Leaving it up to the users to decide which normalisation to apply to any dataset before another operation can be used is very unfriendly and frustrating for them. This is what I can report from a 2-days workshop at the climate office last week. We cannot expect that users know what's in a dataset before they can open it and inspect it. Otherwise we would need to provide a per-ECV handbook with the software that explains which kind of normalisation should be applied to each dataset.

**Comment by forman** [comment]
This is the current situation (as of 07.05.2018):

* `normalize` is never used from our code;
* `normalize_impl` is used 
  * in `EsaCciOdpDataSource.open_dataset` after `open_xarray_dataset` and before `subset_spatial_impl`
  * in `EsaCciOdpDataSource._make_local` after `xr.open_dataset` and before `subset_spatial_impl` 
  * in `LocalDataSource.open_dataset` and `LocalDataSource._make_local` for the same reason as above, btw. with a lot of unbeatiful **code duplication**.
* `adjust_spatial_attrs_impl` is used 
  * as a post-processor in `EsaCciOdpDataSource._make_local` after `subset_spatial_impl`
  * as a post-processor in `LocalDataSource._make_local` after `subset_spatial_impl` (code duplication!)
* `adjust_spatial_attrs` is used 
  * as a post-processor in `coregister` 
  * as a post-processor in `pearson_correlation`
  * as a post-processor in `subset_spatial`
* `adjust_temporal_attrs_impl` is used
  * as a post-processor in `DatasetLike.convert` after converting a DataFrame to Dataset
* `adjust_temporal_attrs` is used 
  * as a post-processor after(?) `normalize` in `open_dataset` and `read_netcdf` if their `normalize` kwarg is set
  * as a post-processor in `temporal_aggregation`
  * as a post-processorin `subset_temporal`

**Comment by kbernat** [comment]
@forman @papesci 
I happy to help if you want, even if it's just a refactoring orremoving some code duplication.
Give me a shout with your thoughts.

**Comment by forman** [comment]
@kbernat Thanks Chris, much appreciated.

**Comment by forman** [comment]
This issue should be fixed after last commit.
